src/java.base/share/classes/java/util/HashMap.java

```diff
@@ -2545,4 +2545,34 @@ else if ((xpl = xp.left) == x) {
         }
     }
 
+    /**
+     * Calculate initial capacity for HashMap based classes, from expected size.
+     *
+     * @param expectedSize expected size
+     * @return initial capacity for HashMap based classes.
+     * @since 19
+     */
+    static int calculateHashMapCapacity(int expectedSize) {
+        if (expectedSize >= 1610612736) {
+            return Integer.MAX_VALUE;
+        }
+        return (expectedSize + (expectedSize + 2) / 3);
+    }
+
+    /**
+     * Creates a new, empty HashMap with an initial table size
+     * accommodating the specified number of elements without the need
+     * to dynamically resize.
```

Comment:
The default load factor should be mentioned.

Reply:
> The default load factor should be mentioned.

@RogerRiggs done.